src/utils/selectors.ts

```diff
@@ -7,13 +8,13 @@ import { Experience } from "types/Firestore";
 import { Purchase } from "types/Purchase";
 import { TextReaction, Reaction, TextReactionType } from "types/reactions";
 import { SparkleSelector } from "types/SparkleSelector";
-import { User } from "types/User";
+import { User, UserWithLocation, userWithLocationToUser } from "types/User";
 import { AnyVenue, PosterPageVenue, VenueEvent } from "types/venues";
 import { ScreeningRoomVideo } from "types/screeningRoom";
 
 import { SovereignVenueState } from "store/reducers/SovereignVenue";
 
-import { WithId } from "utils/id";
+import { withId, WithId } from "utils/id";
```

Comment:
The name of this function should probably start with a verb to make it more obvious that this adds an Id to the passed object.

Reply:
This function has existed for ages and has been used all over the codebase. It's named to match the type that it converts things to, and to 'flow' in a sentence when used (though would do so even better if it was a curried function).

eg. (somewhat contrived examples to show the usage, may not be strictly perfectly valid code)

```typescript
// Currently
const userWithId = withId(user, userId);

const users: User[] = /* etc */;
const usersWithIds = users.map(user => withId(user, userId))

// -----

// If it was written as a curried function
const userWithId = withId(userId)(user);

const users: User[] = /* etc */;
const usersWithIds = users.map(withId(userId))
```

---

Do you have a specific suggestion for a relevant verb to be used for it + arguments for it? It hasn't seemed to be an issue so far, and was introduced back in november apparently:

> commit db093d3b6305f082ccdea171c00149c6731a5756
> Author: Glenn 'devalias' Grant
> Date:   Sat Nov 21 22:25:29 2020 +1100
> 
>     add generic `withId()` helper

Reply:
Definitely OOS of this PR in any case.

Reply:
Marking this as resolved as no action required here IMO.